### ISSUE TITLE
[kubernetes] field mapping exception while ingesting audit logs

### ISSUE THREAD

**Issue by imays11** [elastic/integrations#3664]
I've been using the Kubernetes Audit Log integration for creating Detection Rules since the last update which added several fields. There are still a few fields that would be beneficial for creating rules and for the general analyst experience. Below are a list of fields and their classifications. I've also included an event that has some of these fields (marked with a -). I will add another event to include the rest. 

```
- responseStatus.reason | keyword
- annotations.authorization_k8s_io/decision | keyword
- annotations.authorization_k8s_io/reason | text
- event.dataset | keyword (used to mapped but now it has been removed from the mapping)
- requestObject.spec.serviceAccountName | keyword
- requestObject.spec.restartPolicy | keyword
requestObject.spec.securityContext.runAsUser | long
requestObject.spec.securityContext.runAsGroup | long
requestObject.spec.securityContext.runAsNonRoot | boolean
- requestObject.spec.containers.securityContext.runAsUser | long (‘runasuser’ is mapped with a capitalization error)
requestObject.spec.containers.securityContext.runAsGroup | long
requestObject.spec.containers.securityContext.runAsNonRoot | boolean
requestObject.spec.containers.securityContext.procMount | keyword
requestObject.spec.containers.securityContext.capabilities.add | array? (need to test this and see how it shows up in audit logs)

```

<details>

```
{
  "_index": ".ds-logs-kubernetes.audit_logs-default-2022.06.16-000004",
  "_id": "O8Yiz4EBJHAR6blyEqpi",
  "_version": 1,
  "_score": 1,
  "_source": {
    "kubernetes": {
      "audit": {
        "auditID": "8b8398cd-1d93-4659-805e-0c9d616aa960",
        "requestReceivedTimestamp": "2022-07-05T16:08:04.196323Z",
        "objectRef": {
          "apiVersion": "v1",
          "resource": "pods",
          "namespace": "kube-system"
        },
        "level": "RequestResponse",
        "kind": "Event",
        "verb": "create",
        "annotations": {
          "authorization_k8s_io/decision": "allow",
          "authorization_k8s_io/reason": "RBAC: allowed by ClusterRoleBinding \"system:controller:daemon-set-controller\" of ClusterRole \"system:controller:daemon-set-controller\" to ServiceAccount \"daemon-set-controller/kube-system\""
        },
        "userAgent": "k3s/v1.23.6+k3s1 (linux/amd64) kubernetes/418c3fa/system:serviceaccount:kube-system:daemon-set-controller",
        "requestURI": "/api/v1/namespaces/kube-system/pods",
        "responseStatus": {
          "reason": "Forbidden",
          "metadata": {},
          "code": 403,
          "status": "Failure"
        },
        "stageTimestamp": "2022-07-05T16:08:04.199050Z",
        "responseObject": {
          "reason": "Forbidden",
          "apiVersion": "v1",
          "code": 403,
          "kind": "Status",
          "details": {
            "kind": "pods",
            "name": "elastic-agent-"
          },
          "message": "pods \"elastic-agent-\" is forbidden: error looking up service account kube-system/elastic-agent: serviceaccount \"elastic-agent\" not found",
          "status": "Failure"
        },
        "sourceIPs": [
          "127.0.0.1"
        ],
        "apiVersion": "audit.k8s.io/v1",
        "stage": "ResponseComplete",
        "requestObject": {
          "apiVersion": "v1",
          "kind": "Pod",
          "spec": {
            "dnsPolicy": "ClusterFirstWithHostNet",
            "terminationGracePeriodSeconds": 30,
            "enableServiceLinks": true,
            "hostNetwork": true,
            "serviceAccountName": "elastic-agent",
            "volumes": [
              {
                "name": "proc",
                "hostPath": {
                  "path": "/proc",
                  "type": ""
                }
              },
              {
                "name": "cgroup",
                "hostPath": {
                  "path": "/sys/fs/cgroup",
                  "type": ""
                }
              },
              {
                "name": "varlibdockercontainers",
                "hostPath": {
                  "path": "/var/lib/docker/containers",
                  "type": ""
                }
              },
              {
                "name": "varlog",
                "hostPath": {
                  "path": "/var/log",
                  "type": ""
                }
              }
            ],
            "serviceAccount": "elastic-agent",
            "securityContext": {},
            "restartPolicy": "Always",
            "tolerations": [
              {
                "effect": "NoSchedule",
                "key": "node-role.kubernetes.io/master"
              },
              {
                "effect": "NoExecute",
                "key": "node.kubernetes.io/not-ready",
                "operator": "Exists"
              },
              {
                "effect": "NoExecute",
                "operator": "Exists",
                "key": "node.kubernetes.io/unreachable"
              },
              {
                "effect": "NoSchedule",
                "key": "node.kubernetes.io/disk-pressure",
                "operator": "Exists"
              },
              {
                "effect": "NoSchedule",
                "operator": "Exists",
                "key": "node.kubernetes.io/memory-pressure"
              },
              {
                "effect": "NoSchedule",
                "key": "node.kubernetes.io/pid-pressure",
                "operator": "Exists"
              },
              {
                "effect": "NoSchedule",
                "operator": "Exists",
                "key": "node.kubernetes.io/unschedulable"
              },
              {
                "effect": "NoSchedule",
                "key": "node.kubernetes.io/network-unavailable",
                "operator": "Exists"
              }
            ],
            "containers": [
              {
                "imagePullPolicy": "IfNotPresent",
                "image": "docker.elastic.co/beats/elastic-agent:8.2.3",
                "terminationMessagePolicy": "File",
                "terminationMessagePath": "/dev/termination-log",
                "name": "elastic-agent",
                "resources": {
                  "requests": {
                    "memory": "200Mi",
                    "cpu": "100m"
                  },
                  "limits": {
                    "memory": "500Mi"
                  }
                },
                "securityContext": {
                  "runAsUser": 0
                },
                "env": [
                  {
                    "name": "FLEET_ENROLL",
                    "value": "1"
                  },
                  {
                    "name": "FLEET_INSECURE",
                    "value": "true"
                  },
                  {
                    "name": "FLEET_URL",
                    "value": "https://e78d2f7e27be4550bc6203c622e385d2.fleet.us-central1.gcp.cloud.es.io:443"
                  },
                  {
                    "name": "FLEET_ENROLLMENT_TOKEN",
                    "value": "dHplWVpvQUJoQnA5SHVidTQzbkg6UXE0Rlo0QVlSTm1VdXh4c0pvVVdsUQ=="
                  },
                  {
                    "name": "KIBANA_HOST",
                    "value": "http://kibana:5601"
                  },
                  {
                    "name": "KIBANA_FLEET_USERNAME",
                    "value": "elastic"
                  },
                  {
                    "name": "KIBANA_FLEET_PASSWORD",
                    "value": "changeme"
                  },
                  {
                    "name": "NODE_NAME",
                    "valueFrom": {
                      "fieldRef": {
                        "apiVersion": "v1",
                        "fieldPath": "spec.nodeName"
                      }
                    }
                  },
                  {
                    "name": "POD_NAME",
                    "valueFrom": {
                      "fieldRef": {
                        "apiVersion": "v1",
                        "fieldPath": "metadata.name"
                      }
                    }
                  }
                ],
                "volumeMounts": [
                  {
                    "mountPath": "/hostfs/proc",
                    "name": "proc",
                    "readOnly": true
                  },
                  {
                    "mountPath": "/hostfs/sys/fs/cgroup",
                    "name": "cgroup",
                    "readOnly": true
                  },
                  {
                    "mountPath": "/var/lib/docker/containers",
                    "name": "varlibdockercontainers",
                    "readOnly": true
                  },
                  {
                    "mountPath": "/var/log",
                    "name": "varlog",
                    "readOnly": true
                  }
                ]
              }
            ],
            "schedulerName": "default-scheduler",
            "affinity": {
              "nodeAffinity": {
                "requiredDuringSchedulingIgnoredDuringExecution": {
                  "nodeSelectorTerms": [
                    {
                      "matchFields": [
                        {
                          "values": [
                            "k3d-iammai-cluster2-server-0"
                          ],
                          "key": "metadata.name",
                          "operator": "In"
                        }
                      ]
                    }
                  ]
                }
              }
            }
          },
          "status": {}
        },
        "user": {
          "uid": "13eb3e11-32ca-4dfe-9425-cb12db38daa0",
          "groups": [
            "system:serviceaccounts",
            "system:serviceaccounts:kube-system",
            "system:authenticated"
          ],
          "username": "system:serviceaccount:kube-system:daemon-set-controller"
        }
      }
    },
    "agent": {
      "name": "k3d-iammai-cluster2-agent-0",
      "id": "3f955478-2ceb-48f2-ae34-de031822236e",
      "ephemeral_id": "7b026d30-002d-4b9e-88ac-563f8b1c8205",
      "type": "filebeat",
      "version": "8.2.3"
    },
    "log": {
      "file": {
        "path": "/var/log/kubernetes/audit/audit.log"
      },
      "offset": 6850181
    },
    "elastic_agent": {
      "id": "3f955478-2ceb-48f2-ae34-de031822236e",
      "version": "8.2.3",
      "snapshot": false
    },
    "input": {
      "type": "filestream"
    },
    "@timestamp": "2022-07-05T16:12:12.934Z",
    "ecs": {
      "version": "8.0.0"
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "kubernetes.audit_logs"
    },
    "host": {
      "hostname": "k3d-iammai-cluster2-agent-0",
      "os": {
        "kernel": "5.10.16.3-microsoft-standard-WSL2",
        "codename": "focal",
        "name": "Ubuntu",
        "family": "debian",
        "type": "linux",
        "version": "20.04.4 LTS (Focal Fossa)",
        "platform": "ubuntu"
      },
      "containerized": true,
      "ip": [
        "10.42.1.0",
        "10.42.1.1",
        "172.18.0.3"
      ],
      "name": "k3d-iammai-cluster2-agent-0",
      "mac": [
        "fa:27:bb:86:0c:06",
        "ae:5c:a3:10:4c:e9",
        "b6:63:b9:7d:d0:30",
        "ea:66:37:d2:5a:3c",
        "0e:38:53:76:88:1a",
        "02:42:ac:12:00:03"
      ],
      "architecture": "x86_64"
    },
    "event": {
      "agent_id_status": "verified",
      "ingested": "2022-07-05T16:12:05Z",
      "dataset": "kubernetes.audit_logs"
    }
  },
  "fields": {
    "kubernetes.audit.auditID": [
      "8b8398cd-1d93-4659-805e-0c9d616aa960"
    ],
    "kubernetes.audit.requestObject.spec.containers.image": [
      "docker.elastic.co/beats/elastic-agent:8.2.3"
    ],
    "elastic_agent.version": [
      "8.2.3"
    ],
    "kubernetes.audit.responseStatus.code": [
      403
    ],
    "host.os.name.text": [
      "Ubuntu"
    ],
    "kubernetes.audit.requestObject.spec.enableServiceLinks": [
      true
    ],
    "kubernetes.audit.requestObject.spec.containers.securityContext.runAsUser": [
      0
    ],
    "kubernetes.audit.requestReceivedTimestamp": [
      "2022-07-05T16:08:04.196Z"
    ],
    "kubernetes.audit.requestObject.spec.volumes.name": [
      "proc",
      "cgroup",
      "varlibdockercontainers",
      "varlog"
    ],
    "kubernetes.audit.objectRef.resource": [
      "pods"
    ],
    "host.hostname": [
      "k3d-iammai-cluster2-agent-0"
    ],
    "host.mac": [
      "fa:27:bb:86:0c:06",
      "ae:5c:a3:10:4c:e9",
      "b6:63:b9:7d:d0:30",
      "ea:66:37:d2:5a:3c",
      "0e:38:53:76:88:1a",
      "02:42:ac:12:00:03"
    ],
    "kubernetes.audit.responseObject.kind": [
      "Status"
    ],
    "kubernetes.audit.requestObject.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms.matchFields.key": [
      "metadata.name"
    ],
    "kubernetes.audit.requestObject.spec.tolerations.effect": [
      "NoSchedule",
      "NoExecute",
      "NoExecute",
      "NoSchedule",
      "NoSchedule",
      "NoSchedule",
      "NoSchedule",
      "NoSchedule"
    ],
    "kubernetes.audit.requestObject.spec.containers.terminationMessagePolicy": [
      "File"
    ],
    "host.os.version": [
      "20.04.4 LTS (Focal Fossa)"
    ],
    "kubernetes.audit.userAgent": [
      "k3s/v1.23.6+k3s1 (linux/amd64) kubernetes/418c3fa/system:serviceaccount:kube-system:daemon-set-controller"
    ],
    "kubernetes.audit.annotations.authorization_k8s_io/reason": [
      "RBAC: allowed by ClusterRoleBinding \"system:controller:daemon-set-controller\" of ClusterRole \"system:controller:daemon-set-controller\" to ServiceAccount \"daemon-set-controller/kube-system\""
    ],
    "host.os.name": [
      "Ubuntu"
    ],
    "kubernetes.audit.responseStatus.status": [
      "Failure"
    ],
    "kubernetes.audit.requestObject.apiVersion": [
      "v1"
    ],
    "agent.name": [
      "k3d-iammai-cluster2-agent-0"
    ],
    "host.name": [
      "k3d-iammai-cluster2-agent-0"
    ],
    "event.agent_id_status": [
      "verified"
    ],
    "kubernetes.audit.user.groups": [
      "system:serviceaccounts",
      "system:serviceaccounts:kube-system",
      "system:authenticated"
    ],
    "kubernetes.audit.annotations.authorization_k8s_io/decision": [
      "allow"
    ],
    "host.os.type": [
      "linux"
    ],
    "kubernetes.audit.objectRef.namespace": [
      "kube-system"
    ],
    "kubernetes.audit.responseStatus.reason": [
      "Forbidden"
    ],
    "kubernetes.audit.responseObject.code": [
      403
    ],
    "kubernetes.audit.requestObject.spec.serviceAccountName": [
      "elastic-agent"
    ],
    "kubernetes.audit.requestObject.spec.tolerations.operator": [
      "Exists",
      "Exists",
      "Exists",
      "Exists",
      "Exists",
      "Exists",
      "Exists"
    ],
    "kubernetes.audit.requestObject.spec.volumes.hostPath": [
      {
        "path": "/proc",
        "type": ""
      },
      {
        "path": "/sys/fs/cgroup",
        "type": ""
      },
      {
        "path": "/var/lib/docker/containers",
        "type": ""
      },
      {
        "path": "/var/log",
        "type": ""
      }
    ],
    "kubernetes.audit.responseObject.status": [
      "Failure"
    ],
    "kubernetes.audit.requestObject.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms.matchFields.values": [
      "k3d-iammai-cluster2-server-0"
    ],
    "input.type": [
      "filestream"
    ],
    "kubernetes.audit.sourceIPs": [
      "127.0.0.1"
    ],
    "log.offset": [
      6850181
    ],
    "kubernetes.audit.requestObject.spec.dnsPolicy": [
      "ClusterFirstWithHostNet"
    ],
    "kubernetes.audit.verb": [
      "create"
    ],
    "data_stream.type": [
      "logs"
    ],
    "kubernetes.audit.requestObject.spec.containers.env.value": [
      "1",
      "true",
      "https://e78d2f7e27be4550bc6203c622e385d2.fleet.us-central1.gcp.cloud.es.io:443",
      "dHplWVpvQUJoQnA5SHVidTQzbkg6UXE0Rlo0QVlSTm1VdXh4c0pvVVdsUQ==",
      "http://kibana:5601",
      "elastic",
      "changeme"
    ],
    "host.architecture": [
      "x86_64"
    ],
    "kubernetes.audit.requestObject.kind": [
      "Pod"
    ],
    "agent.id": [
      "3f955478-2ceb-48f2-ae34-de031822236e"
    ],
    "ecs.version": [
      "8.0.0"
    ],
    "host.containerized": [
      true
    ],
    "kubernetes.audit.requestObject.spec.containers.env.valueFrom.fieldRef.fieldPath": [
      "spec.nodeName",
      "metadata.name"
    ],
    "agent.version": [
      "8.2.3"
    ],
    "kubernetes.audit.requestObject.spec.containers.env.name": [
      "FLEET_ENROLL",
      "FLEET_INSECURE",
      "FLEET_URL",
      "FLEET_ENROLLMENT_TOKEN",
      "KIBANA_HOST",
      "KIBANA_FLEET_USERNAME",
      "KIBANA_FLEET_PASSWORD",
      "NODE_NAME",
      "POD_NAME"
    ],
    "host.os.family": [
      "debian"
    ],
    "kubernetes.audit.kind": [
      "Event"
    ],
    "kubernetes.audit.user.username": [
      "system:serviceaccount:kube-system:daemon-set-controller"
    ],
    "kubernetes.audit.requestObject.spec.containers.volumeMounts": [
      {
        "mountPath": "/hostfs/proc",
        "name": "proc",
        "readOnly": true
      },
      {
        "mountPath": "/hostfs/sys/fs/cgroup",
        "name": "cgroup",
        "readOnly": true
      },
      {
        "mountPath": "/var/lib/docker/containers",
        "name": "varlibdockercontainers",
        "readOnly": true
      },
      {
        "mountPath": "/var/log",
        "name": "varlog",
        "readOnly": true
      }
    ],
    "kubernetes.audit.requestObject.spec.containers.terminationMessagePath": [
      "/dev/termination-log"
    ],
    "kubernetes.audit.responseObject.details.kind": [
      "pods"
    ],
    "kubernetes.audit.requestObject.spec.containers.env.valueFrom.fieldRef.apiVersion": [
      "v1",
      "v1"
    ],
    "host.ip": [
      "10.42.1.0",
      "10.42.1.1",
      "172.18.0.3"
    ],
    "kubernetes.audit.responseObject.details.name": [
      "elastic-agent-"
    ],
    "kubernetes.audit.requestObject.spec.schedulerName": [
      "default-scheduler"
    ],
    "agent.type": [
      "filebeat"
    ],
    "host.os.kernel": [
      "5.10.16.3-microsoft-standard-WSL2"
    ],
    "kubernetes.audit.responseObject.apiVersion": [
      "v1"
    ],
    "kubernetes.audit.stageTimestamp": [
      "2022-07-05T16:08:04.199Z"
    ],
    "kubernetes.audit.stage": [
      "ResponseComplete"
    ],
    "kubernetes.audit.apiVersion": [
      "audit.k8s.io/v1"
    ],
    "elastic_agent.snapshot": [
      false
    ],
    "kubernetes.audit.user.uid": [
      "13eb3e11-32ca-4dfe-9425-cb12db38daa0"
    ],
    "kubernetes.audit.requestObject.spec.terminationGracePeriodSeconds": [
      30
    ],
    "kubernetes.audit.requestObject.spec.containers.resources.requests.cpu": [
      "100m"
    ],
    "kubernetes.audit.requestObject.spec.containers.imagePullPolicy": [
      "IfNotPresent"
    ],
    "kubernetes.audit.requestObject.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms.matchFields.operator": [
      "In"
    ],
    "elastic_agent.id": [
      "3f955478-2ceb-48f2-ae34-de031822236e"
    ],
    "data_stream.namespace": [
      "default"
    ],
    "host.os.codename": [
      "focal"
    ],
    "kubernetes.audit.requestObject.spec.containers.resources.requests.memory": [
      "200Mi"
    ],
    "kubernetes.audit.responseObject.message": [
      "pods \"elastic-agent-\" is forbidden: error looking up service account kube-system/elastic-agent: serviceaccount \"elastic-agent\" not found"
    ],
    "kubernetes.audit.responseObject.reason": [
      "Forbidden"
    ],
    "event.ingested": [
      "2022-07-05T16:12:05.000Z"
    ],
    "@timestamp": [
      "2022-07-05T16:12:12.934Z"
    ],
    "kubernetes.audit.requestURI": [
      "/api/v1/namespaces/kube-system/pods"
    ],
    "host.os.platform": [
      "ubuntu"
    ],
    "data_stream.dataset": [
      "kubernetes.audit_logs"
    ],
    "log.file.path": [
      "/var/log/kubernetes/audit/audit.log"
    ],
    "kubernetes.audit.level": [
      "RequestResponse"
    ],
    "agent.ephemeral_id": [
      "7b026d30-002d-4b9e-88ac-563f8b1c8205"
    ],
    "kubernetes.audit.requestObject.spec.restartPolicy": [
      "Always"
    ],
    "kubernetes.audit.requestObject.spec.hostNetwork": [
      true
    ],
    "kubernetes.audit.requestObject.spec.tolerations.key": [
      "node-role.kubernetes.io/master",
      "node.kubernetes.io/not-ready",
      "node.kubernetes.io/unreachable",
      "node.kubernetes.io/disk-pressure",
      "node.kubernetes.io/memory-pressure",
      "node.kubernetes.io/pid-pressure",
      "node.kubernetes.io/unschedulable",
      "node.kubernetes.io/network-unavailable"
    ],
    "kubernetes.audit.requestObject.spec.containers.name": [
      "elastic-agent"
    ],
    "kubernetes.audit.requestObject.spec.serviceAccount": [
      "elastic-agent"
    ],
    "kubernetes.audit.objectRef.apiVersion": [
      "v1"
    ],
    "kubernetes.audit.requestObject.spec.containers.resources.limits.memory": [
      "500Mi"
    ],
    "event.dataset": [
      "kubernetes.audit_logs"
    ]
  }
}

```
</details>

**Comment by gizas** [comment]
As we started having a look, https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#podsecuritycontext-v1-core

RunAsUser and runAsGroup are integers. @imays11 I think we should keep them ints and not long

**Comment by gizas** [comment]
https://github.com/elastic/integrations/blob/fixing-audit-logs/packages/kubernetes/data_stream/audit_logs/fields/fields.yml

First push for above fields above. I will try to run some tests and let you know

**Comment by imays11** [comment]
> As we started having a look, https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#podsecuritycontext-v1-core
> 
> RunAsUser and runAsGroup are integers. @imays11 I think we should keep them ints and not long

Sure thing!

**Comment by tetianakravchenko** [comment]
when building package I am getting error:
```
elastic-package-0.59.0 build
...
data_stream/audit_logs/fields/fields.yml" is invalid: field 0.fields.17.fields.0.name: Does not match pattern '^[\-*_@A-Za-z0-9]+(\.[\-*_@A-Za-z0-9]+)*$'
```
error happens for the fields - `authorization_k8s_io/decision` and `authorization_k8s_io/reason`

**Comment by tetianakravchenko** [comment]
@imays11 I've opened PR - https://github.com/elastic/integrations/pull/4081, please share sample of the log with `requestObject.spec.containers.securityContext.capabilities.add`.

note: `annotations.authorization_k8s_io/decision` and `annotations.authorization_k8s_io/reason` are missing for now, because of the issue described in comment above, I would proceed without it to unblock some issues, depending on this change, and clarify meanwhile when spec change will be available.

**Comment by imays11** [comment]
@tetianakravchenko below is the sample log you've asked for, I hope it is helpful

<details>

```
{
  "_index": ".ds-logs-kubernetes.audit_logs-default-2022.08.30-000001",
  "_id": "Ztf77YIBbqdQ-MxgS1Gx",
  "_version": 1,
  "_score": 0,
  "_source": {
    "kubernetes": {
      "audit": {
        "auditID": "136c8f56-e41b-4c6e-bd05-6fc3b7948f0e",
        "requestReceivedTimestamp": "2022-08-30T09:00:44.046993Z",
        "objectRef": {
          "apiVersion": "v1",
          "resource": "pods",
          "namespace": "default",
          "name": "addcaps"
        },
        "level": "RequestResponse",
        "kind": "Event",
        "verb": "create",
        "annotations": {
          "authorization_k8s_io/decision": "allow",
          "pod-security_kubernetes_io/enforce-policy": "privileged:latest",
          "authorization_k8s_io/reason": ""
        },
        "userAgent": "kubectl1.25.0/v1.25.0 (linux/amd64) kubernetes/a866cbe",
        "requestURI": "/api/v1/namespaces/default/pods?fieldManager=kubectl-client-side-apply&fieldValidation=Strict",
        "responseStatus": {
          "metadata": {},
          "code": 201
        },
        "stageTimestamp": "2022-08-30T09:00:44.050257Z",
        "sourceIPs": [
          "172.22.0.5"
        ],
        "responseObject": {
          "apiVersion": "v1",
          "kind": "Pod",
          "spec": {
            "dnsPolicy": "ClusterFirst",
            "terminationGracePeriodSeconds": 30,
            "enableServiceLinks": true,
            "serviceAccountName": "default",
            "volumes": [
              {
                "projected": {
                  "sources": [
                    {
                      "serviceAccountToken": {
                        "path": "token",
                        "expirationSeconds": 3607
                      }
                    },
                    {
                      "configMap": {
                        "name": "kube-root-ca.crt",
                        "items": [
                          {
                            "path": "ca.crt",
                            "key": "ca.crt"
                          }
                        ]
                      }
                    },
                    {
                      "downwardAPI": {
                        "items": [
                          {
                            "path": "namespace",
                            "fieldRef": {
                              "apiVersion": "v1",
                              "fieldPath": "metadata.namespace"
                            }
                          }
                        ]
                      }
                    }
                  ],
                  "defaultMode": 420
                },
                "name": "kube-api-access-r9wvx"
              }
            ],
            "serviceAccount": "default",
            "securityContext": {},
            "priority": 0,
            "restartPolicy": "Always",
            "preemptionPolicy": "PreemptLowerPriority",
            "tolerations": [
              {
                "effect": "NoExecute",
                "tolerationSeconds": 300,
                "key": "node.kubernetes.io/not-ready",
                "operator": "Exists"
              },
              {
                "effect": "NoExecute",
                "tolerationSeconds": 300,
                "operator": "Exists",
                "key": "node.kubernetes.io/unreachable"
              }
            ],
            "containers": [
              {
                "image": "ollijanatuinen/capsh",
                "imagePullPolicy": "IfNotPresent",
                "terminationMessagePolicy": "File",
                "terminationMessagePath": "/dev/termination-log",
                "name": "addcaps",
                "resources": {},
                "securityContext": {
                  "capabilities": {
                    "add": [
                      "MKNOD",
                      "SYS_ADMIN",
                      "SYS_MODULE",
                      "SYS_RAWIO",
                      "NET_ADMIN",
                      "SYS_CHROOT",
                      "SYS_PTRACE",
                      "NET_RAW",
                      "SYS_BOOT",
                      "SYSLOG",
                      ""
                    ],
                    "drop": [
                      "all"
                    ]
                  }
                },
                "command": [
                  "/bin/sleep",
                  "3650d"
                ],
                "volumeMounts": [
                  {
                    "mountPath": "/var/run/secrets/kubernetes.io/serviceaccount",
                    "name": "kube-api-access-r9wvx",
                    "readOnly": true
                  }
                ]
              }
            ],
            "schedulerName": "default-scheduler"
          },
          "status": {
            "phase": "Pending",
            "qosClass": "BestEffort"
          }
        },
        "apiVersion": "audit.k8s.io/v1",
        "stage": "ResponseComplete",
        "requestObject": {
          "apiVersion": "v1",
          "kind": "Pod",
          "spec": {
            "dnsPolicy": "ClusterFirst",
            "terminationGracePeriodSeconds": 30,
            "enableServiceLinks": true,
            "containers": [
              {
                "imagePullPolicy": "IfNotPresent",
                "image": "ollijanatuinen/capsh",
                "terminationMessagePolicy": "File",
                "terminationMessagePath": "/dev/termination-log",
                "name": "addcaps",
                "resources": {},
                "securityContext": {
                  "capabilities": {
                    "add": [
                      "MKNOD",
                      "SYS_ADMIN",
                      "SYS_MODULE",
                      "SYS_RAWIO",
                      "NET_ADMIN",
                      "SYS_CHROOT",
                      "SYS_PTRACE",
                      "NET_RAW",
                      "SYS_BOOT",
                      "SYSLOG",
                      ""
                    ],
                    "drop": [
                      "all"
                    ]
                  }
                },
                "command": [
                  "/bin/sleep",
                  "3650d"
                ]
              }
            ],
            "securityContext": {},
            "schedulerName": "default-scheduler",
            "restartPolicy": "Always"
          },
          "status": {}
        },
        "user": {
          "groups": [
            "system:masters",
            "system:authenticated"
          ],
          "username": "system:admin"
        }
      }
    },
    "input": {
      "type": "filestream"
    },
    "agent": {
      "name": "k3d-iam-cluster-server-0",
      "id": "cb23b7be-e85b-4dc8-8816-c8c01dba9299",
      "ephemeral_id": "aa03e393-34f8-4f33-a44e-63bfff120a82",
      "type": "filebeat",
      "version": "8.4.0"
    },
    "@timestamp": "2022-08-30T09:00:45.500Z",
    "ecs": {
      "version": "8.0.0"
    },
    "log": {
      "file": {
        "path": "/var/log/kubernetes/kube-apiserver-audit.log"
      },
      "offset": 10000205
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "kubernetes.audit_logs"
    },
    "elastic_agent": {
      "id": "cb23b7be-e85b-4dc8-8816-c8c01dba9299",
      "version": "8.4.0",
      "snapshot": false
    },
    "host": {
      "hostname": "k3d-iam-cluster-server-0",
      "os": {
        "kernel": "5.10.16.3-microsoft-standard-WSL2",
        "codename": "focal",
        "name": "Ubuntu",
        "family": "debian",
        "type": "linux",
        "version": "20.04.4 LTS (Focal Fossa)",
        "platform": "ubuntu"
      },
      "containerized": true,
      "ip": [
        "10.42.0.0",
        "10.42.0.1",
        "172.22.0.2"
      ],
      "name": "k3d-iam-cluster-server-0",
      "mac": [
        "02:42:ac:16:00:02",
        "62:e7:7f:bd:56:1f",
        "92:be:b1:a3:16:2d",
        "b2:f2:95:2f:e7:af",
        "d6:ff:45:0e:61:53"
      ],
      "architecture": "x86_64"
    },
    "event": {
      "agent_id_status": "verified",
      "ingested": "2022-08-30T09:00:45Z",
      "dataset": "kubernetes.audit_logs"
    }
  },
  "fields": {
    "elastic_agent.version": [
      "8.4.0"
    ],
    "host.os.name.text": [
      "Ubuntu"
    ],
    "kubernetes.audit.requestObject.spec.enableServiceLinks": [
      true
    ],
    "host.hostname": [
      "k3d-iam-cluster-server-0"
    ],
    "host.mac": [
      "02:42:ac:16:00:02",
      "62:e7:7f:bd:56:1f",
      "92:be:b1:a3:16:2d",
      "b2:f2:95:2f:e7:af",
      "d6:ff:45:0e:61:53"
    ],
    "kubernetes.audit.responseObject.kind": [
      "Pod"
    ],
    "kubernetes.audit.responseObject.spec.dnsPolicy": [
      "ClusterFirst"
    ],
    "kubernetes.audit.responseObject.spec.volumes.projected.sources.serviceAccountToken.expirationSeconds": [
      3607
    ],
    "host.os.version": [
      "20.04.4 LTS (Focal Fossa)"
    ],
    "kubernetes.audit.responseObject.spec.tolerations.effect": [
      "NoExecute",
      "NoExecute"
    ],
    "kubernetes.audit.requestObject.apiVersion": [
      "v1"
    ],
    "agent.name": [
      "k3d-iam-cluster-server-0"
    ],
    "event.agent_id_status": [
      "verified"
    ],
    "kubernetes.audit.responseObject.spec.enableServiceLinks": [
      true
    ],
    "kubernetes.audit.responseObject.spec.containers.terminationMessagePolicy": [
      "File"
    ],
    "host.os.type": [
      "linux"
    ],
    "kubernetes.audit.responseObject.spec.priority": [
      0
    ],
    "kubernetes.audit.responseObject.spec.containers.securityContext.capabilities.add": [
      "MKNOD",
      "SYS_ADMIN",
      "SYS_MODULE",
      "SYS_RAWIO",
      "NET_ADMIN",
      "SYS_CHROOT",
      "SYS_PTRACE",
      "NET_RAW",
      "SYS_BOOT",
      "SYSLOG",
      ""
    ],
    "input.type": [
      "filestream"
    ],
    "kubernetes.audit.verb": [
      "create"
    ],
    "kubernetes.audit.requestObject.spec.containers.securityContext.capabilities.add": [
      "MKNOD",
      "SYS_ADMIN",
      "SYS_MODULE",
      "SYS_RAWIO",
      "NET_ADMIN",
      "SYS_CHROOT",
      "SYS_PTRACE",
      "NET_RAW",
      "SYS_BOOT",
      "SYSLOG",
      ""
    ],
    "host.architecture": [
      "x86_64"
    ],
    "kubernetes.audit.requestObject.kind": [
      "Pod"
    ],
    "agent.id": [
      "cb23b7be-e85b-4dc8-8816-c8c01dba9299"
    ],
    "host.containerized": [
      true
    ],
    "kubernetes.audit.responseObject.spec.restartPolicy": [
      "Always"
    ],
    "kubernetes.audit.responseObject.spec.containers.volumeMounts": [
      {
        "name": "kube-api-access-r9wvx",
        "mountPath": "/var/run/secrets/kubernetes.io/serviceaccount",
        "readOnly": true
      }
    ],
    "kubernetes.audit.responseObject.spec.volumes.name": [
      "kube-api-access-r9wvx"
    ],
    "kubernetes.audit.responseObject.spec.volumes.projected.sources.configMap.items.key": [
      "ca.crt"
    ],
    "kubernetes.audit.responseObject.spec.containers.image": [
      "ollijanatuinen/capsh"
    ],
    "host.ip": [
      "10.42.0.0",
      "10.42.0.1",
      "172.22.0.2"
    ],
    "kubernetes.audit.requestObject.spec.schedulerName": [
      "default-scheduler"
    ],
    "agent.type": [
      "filebeat"
    ],
    "kubernetes.audit.responseObject.apiVersion": [
      "v1"
    ],
    "kubernetes.audit.responseObject.status.phase": [
      "Pending"
    ],
    "kubernetes.audit.stage": [
      "ResponseComplete"
    ],
    "kubernetes.audit.annotations.pod-security_kubernetes_io/enforce-policy": [
      "privileged:latest"
    ],
    "kubernetes.audit.apiVersion": [
      "audit.k8s.io/v1"
    ],
    "elastic_agent.snapshot": [
      false
    ],
    "kubernetes.audit.requestObject.spec.terminationGracePeriodSeconds": [
      30
    ],
    "kubernetes.audit.responseObject.spec.containers.name": [
      "addcaps"
    ],
    "kubernetes.audit.responseObject.spec.volumes.projected.defaultMode": [
      420
    ],
    "kubernetes.audit.responseObject.spec.volumes.projected.sources.downwardAPI.items.path": [
      "namespace"
    ],
    "elastic_agent.id": [
      "cb23b7be-e85b-4dc8-8816-c8c01dba9299"
    ],
    "host.os.codename": [
      "focal"
    ],
    "kubernetes.audit.responseObject.spec.preemptionPolicy": [
      "PreemptLowerPriority"
    ],
    "kubernetes.audit.responseObject.spec.tolerations.operator": [
      "Exists",
      "Exists"
    ],
    "event.ingested": [
      "2022-08-30T09:00:45.000Z"
    ],
    "@timestamp": [
      "2022-08-30T09:00:45.500Z"
    ],
    "kubernetes.audit.requestURI": [
      "/api/v1/namespaces/default/pods?fieldManager=kubectl-client-side-apply&fieldValidation=Strict"
    ],
    "host.os.platform": [
      "ubuntu"
    ],
    "log.file.path": [
      "/var/log/kubernetes/kube-apiserver-audit.log"
    ],
    "data_stream.dataset": [
      "kubernetes.audit_logs"
    ],
    "kubernetes.audit.level": [
      "RequestResponse"
    ],
    "agent.ephemeral_id": [
      "aa03e393-34f8-4f33-a44e-63bfff120a82"
    ],
    "kubernetes.audit.requestObject.spec.restartPolicy": [
      "Always"
    ],
    "kubernetes.audit.requestObject.spec.containers.name": [
      "addcaps"
    ],
    "kubernetes.audit.auditID": [
      "136c8f56-e41b-4c6e-bd05-6fc3b7948f0e"
    ],
    "kubernetes.audit.requestObject.spec.containers.command": [
      "/bin/sleep",
      "3650d"
    ],
    "kubernetes.audit.requestObject.spec.containers.image": [
      "ollijanatuinen/capsh"
    ],
    "kubernetes.audit.responseStatus.code": [
      201
    ],
    "kubernetes.audit.requestReceivedTimestamp": [
      "2022-08-30T09:00:44.046Z"
    ],
    "kubernetes.audit.objectRef.resource": [
      "pods"
    ],
    "kubernetes.audit.requestObject.spec.containers.terminationMessagePolicy": [
      "File"
    ],
    "kubernetes.audit.userAgent": [
      "kubectl1.25.0/v1.25.0 (linux/amd64) kubernetes/a866cbe"
    ],
    "kubernetes.audit.annotations.authorization_k8s_io/reason": [
      ""
    ],
    "host.os.name": [
      "Ubuntu"
    ],
    "host.name": [
      "k3d-iam-cluster-server-0"
    ],
    "kubernetes.audit.requestObject.spec.containers.securityContext.capabilities.drop": [
      "all"
    ],
    "kubernetes.audit.responseObject.spec.containers.securityContext.capabilities.drop": [
      "all"
    ],
    "kubernetes.audit.user.groups": [
      "system:masters",
      "system:authenticated"
    ],
    "kubernetes.audit.annotations.authorization_k8s_io/decision": [
      "allow"
    ],
    "kubernetes.audit.objectRef.namespace": [
      "default"
    ],
    "kubernetes.audit.responseObject.spec.containers.command": [
      "/bin/sleep",
      "3650d"
    ],
    "kubernetes.audit.sourceIPs": [
      "172.22.0.5"
    ],
    "log.offset": [
      10000205
    ],
    "kubernetes.audit.requestObject.spec.dnsPolicy": [
      "ClusterFirst"
    ],
    "data_stream.type": [
      "logs"
    ],
    "kubernetes.audit.responseObject.spec.schedulerName": [
      "default-scheduler"
    ],
    "ecs.version": [
      "8.0.0"
    ],
    "kubernetes.audit.responseObject.spec.serviceAccount": [
      "default"
    ],
    "agent.version": [
      "8.4.0"
    ],
    "host.os.family": [
      "debian"
    ],
    "kubernetes.audit.responseObject.spec.volumes.projected.sources.downwardAPI.items.fieldRef.fieldPath": [
      "metadata.namespace"
    ],
    "kubernetes.audit.kind": [
      "Event"
    ],
    "kubernetes.audit.user.username": [
      "system:admin"
    ],
    "kubernetes.audit.responseObject.status.qosClass": [
      "BestEffort"
    ],
    "kubernetes.audit.objectRef.name": [
      "addcaps"
    ],
    "kubernetes.audit.requestObject.spec.containers.terminationMessagePath": [
      "/dev/termination-log"
    ],
    "kubernetes.audit.responseObject.spec.terminationGracePeriodSeconds": [
      30
    ],
    "kubernetes.audit.responseObject.spec.containers.terminationMessagePath": [
      "/dev/termination-log"
    ],
    "host.os.kernel": [
      "5.10.16.3-microsoft-standard-WSL2"
    ],
    "kubernetes.audit.stageTimestamp": [
      "2022-08-30T09:00:44.050Z"
    ],
    "kubernetes.audit.responseObject.spec.volumes.projected.sources.serviceAccountToken.path": [
      "token"
    ],
    "kubernetes.audit.requestObject.spec.containers.imagePullPolicy": [
      "IfNotPresent"
    ],
    "data_stream.namespace": [
      "default"
    ],
    "kubernetes.audit.responseObject.spec.volumes.projected.sources.configMap.name": [
      "kube-root-ca.crt"
    ],
    "kubernetes.audit.responseObject.spec.tolerations.key": [
      "node.kubernetes.io/not-ready",
      "node.kubernetes.io/unreachable"
    ],
    "kubernetes.audit.responseObject.spec.volumes.projected.sources.configMap.items.path": [
      "ca.crt"
    ],
    "kubernetes.audit.responseObject.spec.containers.imagePullPolicy": [
      "IfNotPresent"
    ],
    "kubernetes.audit.responseObject.spec.tolerations.tolerationSeconds": [
      300,
      300
    ],
    "kubernetes.audit.objectRef.apiVersion": [
      "v1"
    ],
    "kubernetes.audit.responseObject.spec.volumes.projected.sources.downwardAPI.items.fieldRef.apiVersion": [
      "v1"
    ],
    "event.dataset": [
      "kubernetes.audit_logs"
    ],
    "kubernetes.audit.responseObject.spec.serviceAccountName": [
      "default"
    ]
  }
}

```
</details>

**Comment by imays11** [comment]
> note: `annotations.authorization_k8s_io/decision` and `annotations.authorization_k8s_io/reason` are missing for now, because of the issue described in comment above, I would proceed without it to unblock some issues, depending on this change, and clarify meanwhile when spec change will be available.

@tetianakravchenko Will the PR to fix it - https://github.com/elastic/package-spec/pull/402 - make those fields available along with the rest? Or will you have to open a new PR for them?

**Comment by tetianakravchenko** [comment]
> @tetianakravchenko below is the sample log you've asked for, I hope it is helpful

@imays11 thank you it was rally helpful!

> @tetianakravchenko Will the PR to fix it - https://github.com/elastic/package-spec/pull/402 - make those fields available along with the rest? Or will you have to open a new PR for them?

no, those fields with `/` should be added after new spec version will be released

**Comment by imays11** [comment]
> @imays11 the latest kubernetes package was promoted, it should be soon available

Thank you!
